substrate/frame/election-provider-multi-block/src/signed/tests.rs

```diff
@@ -633,4 +638,250 @@ mod e2e {
 			assert_ok!(Submissions::<Runtime>::ensure_killed(0));
 		})
 	}
+
+	#[test]
+	fn missing_pages_treated_as_empty() {
+		// Test the scenario where a solution has N pages but only some are submitted.
+		// The key behavior being tested: missing pages should be treated as empty/default
+		// rather than causing DataUnavailable errors or verification failures.
+		//
+		// This ensures that:
+		// 1. SolutionDataProvider::get_page() returns Some(default_solution) for missing pages
+		// 2. The verifier processes missing pages as empty (0 winners)
+		// 3. No VerificationDataUnavailable events are emitted for missing pages
+		ExtBuilder::signed().build_and_execute(|| {
+			roll_to_signed_open();
+			assert_full_snapshot();
+
+			// Register with a score for a minimal solution
+			let minimal_score =
+				ElectionScore { minimal_stake: 1, sum_stake: 1, sum_stake_squared: 1 };
+			assert_ok!(SignedPallet::register(RuntimeOrigin::signed(99), minimal_score));
+			assert_eq!(balances(99), (95, 5)); // deposit taken
+
+			// Submit only page 0 with a default/empty solution, deliberately skip pages 1 and 2
+			// This simulates a scenario where a submitter doesn't submit all pages
+			assert_ok!(SignedPallet::submit_page(
+				RuntimeOrigin::signed(99),
+				0,
+				Some(Box::new(Default::default()))
+			));
+
+			// Pages 1 and 2 are deliberately not submitted - they should be treated as
+			// empty/default
+
+			// Verify the metadata shows correct page submission status
+			let metadata = Submissions::<Runtime>::metadata_of(0, 99).unwrap();
+			assert_eq!(metadata.pages.into_inner(), vec![true, false, false]); // only page 0 submitted
+
+			// Verify that the solution data provider can access all pages without errors
+			assert!(SignedPallet::get_page(0).is_some(), "Submitted page 0 should return Some");
+			assert!(
+				SignedPallet::get_page(1).is_some(),
+				"Missing page 1 should return Some with default solution"
+			);
+			assert!(
+				SignedPallet::get_page(2).is_some(),
+				"Missing page 2 should return Some with default solution"
+			);
+
+			// Start verification
+			roll_to_signed_validation_open();
+
+			// The verification should proceed and treat the missing pages as empty
+			roll_next(); // Process page 2 (empty)
+			roll_next(); // Process page 1 (empty)
+			roll_next(); // Process page 0
+
+			// Check that verification handled the missing pages gracefully
+			let verifier_events = verifier_events();
```

Comment:
I think the syntax might be easier to read if you `assert_eq!(verifier_events(), vec![...])`, no?

Reply:
For example with `iter().any()` the order is lost.